src/main/java/org/radarcns/phone/PhoneSensorManager.java

```diff
@@ -104,6 +137,30 @@ public void start(@NonNull final Set<String> acceptableIds) {
             logger.warn("Phone Light sensor not found");
         }
 
+        // Gyroscope
+        if (sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE) != null) {
+            Sensor gyroscope = sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE);
+            sensorManager.registerListener(this, gyroscope, SENSOR_DELAY_DEFAULT);
+        } else {
+            logger.warn("Phone Gyroscope not found");
+        }
+
+        // Magnetic field
+        if (sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD) != null) {
+            Sensor magneticField = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+            sensorManager.registerListener(this, magneticField, SENSOR_DELAY_DEFAULT);
+        } else {
+            logger.warn("Phone magnetometer not found");
+        }
+
+        // Steps
+        if (sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER) != null) {
+            Sensor stepCounter = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER);
```

Comment:
As per <https://developer.android.com/reference/android/hardware/Sensor.html#TYPE_STEP_COUNTER>, the step counter should be registered but not deregistered. Perhaps we could remove the `unregister` call in `close()`, or the step counter listener could be a separate class that is not unregistered?